### ISSUE TITLE
UX - buttons: copy at & copy to clipboard

### ISSUE THREAD

**Issue by Blind4Basics** [hobovsky/katasniplib#19]
**Request:**
currently, there are a lot of clicks to do on user's side. For 2 lines snippets, this means there are no storng advantage to go through the snippets provider rather than just type the code. Here are some ideas to make the snippets provider quicker to use.

### Copy to clipboard button
_(currently automatically inherited from polyglot)_

A button to be shown on each snippet code block

### "Paste at cursor position" button

A second button, on each code block, that would paste the snippet directly at the current cursor position in the "parent" editor.
Things to consider:
- add in the current line? (meaning, possibly right in the middle of the current instruction)
- replace current line?
- insert the snippet before or after current line
- change the behavior if a selection exists? (=> replace the content with the snippet, in that case?)

"insert before current line" would have my preference, I believe (without selection).

_Note: the context menu I was talking about in #14 could be of use to provide different options there. Especially, insert before or replace_

### "Paste at cursor position" on double click

Same as above, but triggering it with a double click on the snippet code block. Most likely more useful than button ('cause, quicker), but the button still would be nice, to showcase the functionality (sort of...).

**Comment by hobovsky** [comment]
> For 2 lines snippets, this means there are no storng advantage to go through the snippets provider rather than just type the code.

One of the ideas behind the library is to have snippets at hand if you do not know how to type them in :) For example, I notoriously miss the name of `std:mt19937` random generator in C++, or get the syntax of `std::mt19937 rand { std::random_device {} () };` wrong because I confuse parentheses with braces, mistype some names, etc. I think that main strength of short snippets is to provide knowledge rather than save keystrokes.

But still the ideas are nice. I did not think of a double click, and it sounds cool. Thanks!

**Comment by Blind4Basics** [comment]
yeah true, but as a mainly python/JS user, I'm mainly thinking about how to make it comfortable for _me_... XD

**Comment by hobovsky** [comment]
How to get access to a CodeMirror editor: `document.querySelector('.CodeMirror').CodeMirror.replaceSelection("asd")`
CodeMirror 5 editor commands: https://codemirror.net/5/doc/manual.html#commands

**Comment by hobovsky** [comment]
Very basic support for "insert selection at cursor" has been added. Now we can think of other options and where to add them.
